halo2_proofs/src/poly/evaluator.rs

```diff
@@ -541,3 +613,61 @@ impl BasisOps for ExtendedLagrangeCoeff {
         domain.rotate_extended(poly, rotation)
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use std::iter;
+
+    use pasta_curves::pallas;
+
+    use super::{get_chunk_params, new_evaluator, Ast, BasisOps, Evaluator};
+    use crate::{
+        multicore,
+        poly::{Coeff, EvaluationDomain, ExtendedLagrangeCoeff, LagrangeCoeff},
+    };
+
+    #[test]
+    fn short_chunk_regression_test() {
+        // Pick the smallest polynomial length that is guaranteed to produce a short chunk
+        // on this machine.
+        let k = match (1..31)
```

Comment:
Maybe this should be smaller so that we prefer to not run the test over testing with an unreasonably large circuit.
```suggestion
        let k = match (1..16)
```